spock-core/src/main/java/spock/mock/MockMakerId.java

```diff
@@ -0,0 +1,16 @@
+package spock.mock;
+
+import org.spockframework.mock.runtime.IMockMaker;
+
+/**
+ * Provides constants for known {@link IMockMaker} implementations.
+ */
+public final class MockMakerId {
```

Comment:
🤔 I'm debating whether we should make this a value class instead.

Reply:
I'm not getting that. What do you mean by value class in this context?

Reply:
So that it has a `String id` field, and you'd use instances of this class instead of plain `String` everywhere. This would also solve the consistent error message problem as you could have a proper `toString()` here.

Reply:
I have changed the class to `MockMakers` and now holds `IMockMakerSettings` objects for each element to allow custom settings to be passed to each `Mock` call, see #1753 for usage details.

Sample:

```groovy
//Without settings:
Subscriber subscriber = Mock(mockMaker: MockMakers.byteBuddy)

//With settings
Subscriber subscriber = Mock(mockMaker: MockMakers.mockitoInline { org.mockito.MockSettings settings ->
      settings.serializable()
    })
````